pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -77,6 +79,10 @@ func (r Requirements) Zones() sets.String {
 	return r.Requirement(v1.LabelTopologyZone)
 }
 
+func (r Requirements) Regions() sets.String {
+	return r.Requirement(v1.LabelTopologyRegion)
```

Comment:
IIUC, since this is nil, it means that region is unconstrained? What if you specify an incorrect region. Does it behave correctly?

Reply:
It does now by failing in the CreateInstance func at runtime.  Before my last push it was a noop.  I don't know a usecase where a region would be specified that isn't where Karpenter is deployed, but I suppose we still shouldn't provision if they are conflicting.